### PR TITLE
refactor: reown

### PR DESCRIPTION
**Pull request by andreabadesso**
### Acceptance Criteria

- We should use `@reown/walletkit` instead of `@walletconnect/web3wallet`
- We should rename all instances of "wallet connect" to "reown"

### Security Checklist
- [X] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,31 +22,32 @@
...
     "@react-native-firebase/messaging": "16.7.0",
     "@react-navigation/bottom-tabs": "6.5.8",
     "@react-navigation/native": "6.1.7",
     "@react-navigation/stack": "6.3.17",
-    "@sentry/react-native": "5.6.0",
-    "@walletconnect/core": "^2.15.1",
-    "@walletconnect/react-native-compat": "^2.12.2",
-    "@walletconnect/web3wallet": "^1.14.1",
+    "@reown/walletkit": "1.1.0",
+    "@sentry/react-native": "5.31.0",
```
Comment (andreabadesso) [inline package.json:38]: This is unrelated to the walletkit refactor, but I had to upgrade it to make it work in the latest XCode

File: package.json [hunk 2]
```diff
@@ -22,31 +22,31 @@
...
+    "@walletconnect/core": "2.17.0",
```
Comment (andreabadesso) [inline package.json:39]: Reown crashes on previous version

They didn't move it to the reown namespace yet

File: locale/pt-br/texts.po [hunk 3]
```diff
@@ -833,65 +831,72 @@ msgstr "Eu quero desregistrar o token **${ tokenLabel }**"
...
+msgid "Sign Message Request"
+msgstr "Solicitação de Assinatura de Mensagem"
+
+#: src/screens/Reown/SignOracleDataRequestScreen.js:25
+msgid "Sign Oracle Data Request"
+msgstr "Solicitação de Assinatura de Dados do Oracle"
+
+#: src/screens/Reown/SuccessFeedbackScreen.js:36
+msgid "Back"
+msgstr ""
```
Comment (alexruzenhack) [inline locale/pt-br/texts.po:898]:
```suggestion
msgstr "Voltar"
```

File: locale/pt-br/texts.po [hunk 4]
```diff
@@ -1382,268 +1389,305 @@ msgstr ""
...
 
-#: src/components/WalletConnect/SignMessageRequest.js:45
+#: src/components/Reown/SignMessageRequest.js:45
 msgid "Address Path"
 msgstr "Path do Endereço"
 
-#: src/components/WalletConnect/WarnDisclaimer.js:27
+#: src/components/Reown/SignOracleDataRequest.js:35
+msgid "Oracle data to sign"
+msgstr ""
```
Comment (alexruzenhack) [inline locale/pt-br/texts.po:1528]: [question] Shouldn't it be translated?

File: locale/pt-br/texts.po [hunk 5]
```diff
@@ -1382,268 +1389,305 @@ msgstr ""
...
 
-#: src/components/WalletConnect/NanoContract/NewNanoContractTransactionModal.js:60
+#: src/components/Reown/NanoContract/NewNanoContractTransactionModal.js:61
 msgid "Review transaction details"
 msgstr "Revisar detalhes da transação"
 
-#: src/components/WalletConnect/NanoContract/NewNanoContractTransactionRequest.js:197
+#: src/components/Reown/NanoContract/NewNanoContractTransactionRequest.js:205
+msgid "Success!"
+msgstr ""
```
Comment (alexruzenhack) [inline locale/pt-br/texts.po:1633]:
```suggestion
msgstr "Sucesso!"
```

File: locale/pt-br/texts.po [hunk 6]
```diff
@@ -1382,268 +1389,305 @@ msgstr ""
...
+
+#: src/components/Reown/NanoContract/NewNanoContractTransactionRequest.js:206
+msgid "Transaction successfully sent."
+msgstr "Transação enviada com sucesso."
+
+#: src/components/Reown/NanoContract/NewNanoContractTransactionRequest.js:244
+msgid "Registering Nano Contract."
+msgstr "Registrando Nano Contract"
```
Comment (alexruzenhack) [inline locale/pt-br/texts.po:1641]:
```suggestion
msgstr "Registrando Nano Contract."
```

File: src/actions.js [hunk 7]
```diff
@@ -867,21 +867,21 @@ export const sharedAddressUpdate = (lastSharedAddress, lastSharedIndex) => ({
...
-  payload: walletConnect,
+  payload: reown,
 });
 
 /**
- * Dispatched with data when a WalletConnect QRCode is read
+ * Dispatched with data when a Reown QRCode is read
  *
- * data {string} The WalletConnect v2 URI
+ * data {string} The Reown v2 URI
```
Comment (alexruzenhack) [inline src/actions.js:882]: [question] The version API is still the same v2?

File: src/components/Reown/ApproveRejectModal.js
Comment (alexruzenhack) [inline src/components/Reown/ApproveRejectModal.js:1]: [note] I believe we shouldn't name our submodules with third party names. This feature could have received its own name to serve as an abstraction between our interpretation of the feature and what we are using underneath to implement it. This strategy can also make it a starting point for a generalization.

**Review by alexruzenhack — APPROVED** [review]

File: locale/pt-br/texts.po (hunk 5 above)
Comment (andreabadesso) [reply locale/pt-br/texts.po]: Fixed, thanks!

File: locale/pt-br/texts.po (hunk 6 above)
Comment (andreabadesso) [reply locale/pt-br/texts.po]: Fixed! Thanks

File: src/actions.js (hunk 7 above)
Comment (andreabadesso) [reply src/actions.js]: No it's not, fixed! Thanks

File: src/components/Reown/ApproveRejectModal.js
Comment (andreabadesso) [reply src/components/Reown/ApproveRejectModal.js:1]: I agree

Comment (andreabadesso) [reply src/components/Reown/ApproveRejectModal.js:1]: Not renaming it in this PR though, it's outside of its scope

Comment (alexruzenhack) [reply src/components/Reown/ApproveRejectModal.js:1]: No problem. Let address it later, but it would be nice to have a tracked issue.

**Review by alexruzenhack — APPROVED** [review]

File: shim.js [hunk 8]
```diff
@@ -43,3 +43,7 @@ if (typeof atob === 'undefined') {
     return Buffer.from(b64Encoded, 'base64').toString('binary');
   };
 }
+
+const { URLSearchParams } = require('react-native-url-polyfill');
+
+global.URLSearchParams = URLSearchParams;
```
Comment (pedroferreira1) [inline shim.js:49]: Do we use this in this repo?

File: locale/ru-ru/texts.po [hunk 9]
```diff
@@ -1288,308 +1295,344 @@ msgstr ""
...
+msgid "Success!"
+msgstr ""
+
+#: src/components/Reown/NanoContract/NewNanoContractTransactionRequest.js:206
+msgid "Transaction successfully sent."
+msgstr ""
+
+#: src/components/Reown/NanoContract/NewNanoContractTransactionRequest.js:244
+#, fuzzy
+msgid "Registering Nano Contract."
```
Comment (pedroferreira1) [inline locale/ru-ru/texts.po:1591]: What's this? This translation didn't exist

File: package.json [hunk 10]
```diff
@@ -22,31 +22,31 @@
...
     "crypto-js": "4.1.1",
     "deprecated-react-native-prop-types": "2.3.0",
-    "ethers": "^6.13.2",
-    "events": "^3.3.0",
-    "fast-text-encoding": "^1.0.6",
-    "hathor-rpc-handler-test": "^0.0.42",
+    "ethers": "6.13.3",
+    "events": "3.3.0",
+    "fast-text-encoding": "1.0.6",
+    "hathor-rpc-handler-test": "0.0.42",
```
Comment (pedroferreira1) [inline package.json:49]: What's this?

File: shim.js (hunk 8 above)
Comment (andreabadesso) [reply shim.js:49]: No, but reown uses it, crashes without this

File: package.json (hunk 10 above)
Comment (andreabadesso) [reply package.json]: This library has been removed and replaced by our namespaced one, I forgot to remove it from package.json, thanks!

Fixed

**Review by pedroferreira1 — APPROVED** [review]

**Pull request merged**